tests/test_audiofile.py

```diff
@@ -1209,6 +1209,82 @@ def test_read_duration_and_offset_file_formats(tmpdir):
             atol=tolerance('duration', sampling_rate),
         )
 
+        # Duration that results in empty signal
+        duration = 0.000001
+        sig, fs = af.read(file, duration=duration)
+        np.testing.assert_array_equal(
+            sig,
+            np.array([], np.float32),
+        )
+
+
+@pytest.mark.parametrize(
+    # offset and duration need to be given in seconds
+    'offset, duration, expected',
+    [
+        (0.1, 0.1, [.1]),
+        (0.1, 0.03, []),
+        (0.1, 0.15, [.1, .2]),
+        (0.1, 0.19, [.1, .2]),
+        (0.049, 0.19, [.0, .1]),
+        (0.15, 0.15, [.2, .3]),
+    ]
+)
+def test_read_duration_and_offset_rounding(tmpdir, offset, duration, expected):
+
+    # Prepare signals
+    sampling_rate = 10
+    signal = np.array(
+        [[.0, .1, .2, .3, .4, .5, .6, .7, .8, .9]],
+        dtype=np.float32,
+    )
+    file = str(tmpdir.join('signal.wav'))
```

Comment:
As the file content is deterministic, would it make sense to pull the whole test wav file generation into a separate fixture?

Reply:
@ChristianGeng are you fine with my implementation of the fixture?